demo/read-n-demo.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+read-one () {
+	# try typing a letter without hitting newline
```

Comment:
Please be consistent with 2 space indents everywhere.

Which editor do you use?  If vim, maybe we can add "modelines" so things automatically work.  I use vim.

Reply:
I use vim. Sorry this keeps being an issue, I just fixed it by putting configuration for this directory in my global vimrc (https://stackoverflow.com/a/18933951/7669110). Shouldn't happen again.